src/mscorlib/shared/System/Diagnostics/StackFrame.cs

```diff
@@ -0,0 +1,272 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System.Text;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Diagnostics.Contracts;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// There is no good reason for the methods of this class to be virtual.
+    /// </summary>
+    public partial class StackFrame
+    {
+        private MethodBase method;
+        private int offset;
+        private int ILOffset;
+        private String strFileName;
+        private int iLineNumber;
+        private int iColumnNumber;
+
+        [System.Runtime.Serialization.OptionalField]
```

Comment:
This class isn't serializable, it is? If so, we shouldn't need OptionalField attributes.